src/OrchardCore.Modules/OrchardCore.Shortcodes/Services/ShortcodeService.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Shortcodes;
+
+namespace OrchardCore.Shortcodes.Services
+{
+    public class ShortcodeService : IShortcodeService
+    {
+        private readonly ShortcodesProcessor _shortcodesProcessor;
+        private readonly IEnumerable<IShortcodeProvider> _shortcodeProviders;
+
+        public ShortcodeService(IEnumerable<IShortcodeProvider> shortcodeProviders)
+        {
+            _shortcodeProviders = shortcodeProviders;
+            _shortcodesProcessor = new ShortcodesProcessor(shortcodeProviders);
+
```

Comment:
Remove extra line

Reply:
And I removed need to store the providers locally, as they go straight to the processor.

Reply:
Cool, didn't see it ;)